app/views/admin/display_ads/index.html.erb

```diff
@@ -1,41 +1,56 @@
-<nav class="flex mb-4" aria-label="Display Ads navigation">
-  <%= form_tag(admin_display_ads_path, method: "get") do %>
-    <%= text_field_tag(:search, params[:search], aria: { label: "Search" }, class: "crayons-header--search-input crayons-textfield", placeholder: "Search by Org name") %>
-  <% end %>
-  <div class="ml-auto">
-    <div class="justify-content-end">
-      <%= link_to "Make A New Display Ad", new_admin_display_ad_path, class: "crayons-btn" %>
+<div
```

Comment:
Refactor (blocking): Instead of creating the markup for a modal every time we need one, create a [ViewComponent](https://viewcomponent.org/) instead.

Badge achievements are currently using the confirmation modal, and it's going to be required for other sections of the admin as well.

https://github.com/forem/forem/blob/dd02f60bc2e4df0b1df1c1e7ee50c2f663526fcc/app/views/admin/badge_achievements/index.html.erb#L1-L7

I haven't worked with ViewComponents yet, but they appear to have good documentation though.

Thoughts @rhymes?

Reply:
> Instead of creating the markup for a modal every time we need one

I'm not sure what you mean by this; do you mean I should not use the `destroy_confirmation_modal` partial? And instead build the modal with ViewComponents?

When I started working on this RFC, I used the modal located in the `adminModal.js` file. But that modal was not accessible (which will not do), so you and Suzanne suggested that I use the Preact Modal component, which I believe is what I have done (by extending `ModalController` from `ConfirmationModalController`). Is the approach I've been using now no longer appropriate?

The PRs I have worked on with my current approach are:
- [Badge Achievements Destroy](https://github.com/forem/forem/pull/14684)
- [Broadcast Destroy](https://github.com/forem/forem/pull/14819)

Reply:
The destroy confirmation modal partial is good. No need to change that. What I meant was what I linked to. Instead of having to always define the `data-confirmation*` attributes on that div, we could just have a ViewComponent to wrap all that and then the rest of the stuff I mentioned above. That way it's all encapsulated.

Reply:
Ahhhh ok I see. I will look into that then.

Reply:
@nickytonline I've looked through the docs for ViewComponents, and it's very interesting. 😄 

Seeing as the `data-controller` code is the same for all destructive actions, it would be a straightforward refactor.

However, I prefer to keep the positive momentum I have built refactoring the destructive actions. This approach works for me. Having completed what I'm working on now, I can then switch contexts, take time to learn view_components, and incorporate it.

I hope this is alright with you, Nick. Please let me know, thanks!

@rhymes I'll also wait for you to weigh-in 👍🏾

Reply:
I think this makes sense @msarit